stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/UserStateService.kt

```diff
@@ -44,6 +44,8 @@ internal class UserStateService {
             onEvent<UserStateEvent.UnsetUser> { _, _ -> UserState.NotSet }
         }
         state<UserState.Anonymous.AnonymousUserSet> {
+            onEvent<UserStateEvent.ConnectAnonymous> { _, _ -> UserState.Anonymous.Pending }
```

Comment:
The same: there is no reason to connect user if we already set anonymous user

Reply:
If the app is killed by the OS, the websocket will lose the connection and enter in this state. A user that sets the connection in an activity will see a crash if he/she doesn't manage it in its own side. Maybe this is a state that shouldn't happen... but it seams too much that the app crashes if the user asks for another connection when the app comes back in the foreground.

Maybe we should deal log it or deal with it in another way? Let's not throw an exception here.

Reply:
If app got killed then nothing is kept in memory, state of this would be Idle

Reply:
Or maybe we do throw an exception, but provide a message, like: 

"You are trying to connection anonymously, but the user is already set, perhaps you are asking for a connection a second time? Please confirm if you are managing the lifecycle of your Activity correctly"

Reply:
I think it's better to check state of `UserStateService` in `ChatClient::setAnonymousUser` similarly to `ChatClient::setUser`